mempool/clist_mempool.go

```diff
@@ -608,7 +616,25 @@ func (mem *CListMempool) Update(
 
 	// Recheck txs left in the mempool to remove them if they became invalid in the new state.
 	if mem.config.Recheck {
-		mem.recheckTxs()
+		// Don't perform recheck if one of the peers is known to be at least at our height +
+		// maxHeightDiffForRechecking.
+		doRecheck := true
+		if mem.sw != nil {
+			mem.sw.Peers().ForEach(func(peer p2p.Peer) {
+				if peerState, ok := peer.Get(types.PeerStateKey).(PeerState); ok {
+					if peerState.GetHeight() >= height+maxHeightDiffForRechecking {
+						doRecheck = false
+						return
+					}
+				}
+			})
+		}
+
+		if doRecheck {
+			mem.recheckTxs()
+		} else {
+			mem.logger.Debug("recheck disabled at this height", "height", height)
```

Comment:
This condition for recheck=false is when node is behind by a max based on the logic above so the debug message could be more related to that too

Reply:
And this should be an INFO message.